spec/controllers/queue_items_controller_spec.rb

```diff
@@ -3,200 +3,173 @@
 describe QueueItemsController do
   describe 'GET index' do
     it 'sets @queue_items with authentication' do
-      user = Fabricate(:user)
-      session[:user_id] = user.id
-      queue_items = user.queue_items
+      set_current_user
+      queue_items = current_user.queue_items
       get :index
       expect(assigns(:queue_items)).to match_array(queue_items)
     end
 
-    it 'redirect_to sign in page with unquthentication' do
-      get :index
-      expect(response).to redirect_to sign_in_path
+    it_behaves_like 'require_sign_in' do
+      let(:action) { get :index }
     end
   end
 
   describe 'POST create' do
-    it 'redirects to the my queue page' do
-      session[:user_id] = Fabricate(:user).id
-      post :create, video_id: Fabricate(:video).id
-      expect(response).to redirect_to my_queue_path
-    end
+    before { set_current_user }
+
+    it_behaves_like 'queue_done' do
+      let(:action) { post :create, video_id: Fabricate(:video).id }
+    end    
 
     it 'create a queue item' do
-      session[:user_id] = Fabricate(:user).id
       post :create, video_id: Fabricate(:video).id
       expect(QueueItem.count).to eq(1)
     end
 
     it 'creates the queue item associated with video' do
-      session[:user_id] = Fabricate(:user).id
       video = Fabricate(:video)
```

Comment:
You can refactor video to let(:video) { Fabricate(:video) } and just reference video in many lines :)